gufe/protocols/protocoldag.py

```diff
@@ -402,23 +401,16 @@ def execute_DAG(protocoldag: ProtocolDAG, *,
         # `ProtocolUnitResult`
         inputs = _pu_to_pur(unit.inputs, results)
 
-        if scratch_basedir is not None:
```

Comment:
@dotsdl the logic here seemed wrong.  If `scratch_basedir is not None`, then you probably don't want to wrap it as a `TemporaryDirectory` if you then want to keep it.

Also, `ProtocolUnit.execute()` **also** has logic for dealing with scratch=None, so we can omit the second branch.